draft-ietf-quic-load-balancers.md

```diff
@@ -384,54 +384,116 @@ beginning with the second octet. These bytes represent the server ID.
+
+The Low-Config CID Algorithm is similar to the plaintext CID algorithm, but
+allows the load balancer to assign server IDs dynamically so that configuration
+does not require bespoke server ID assignment. This also reduces linkability.
+However, it requires state at the load balancer that roughly scales with the
+number of connections, until the server ID codespace is exhausted.
+
+~~~
+Low-Config CID {
+  First Octet (8),
+  Server ID (8..56),
+  For Server Use (0..152-len(Server ID)),
+}
+~~~
 {: #lowconfig-cid-format title="Low-Config CID Format"}
 
 ### Configuration Agent Actions
 
-The configuration agent selects a length for the server ID encoding. This
-length MUST have enough entropy to have a different code point for each server.
+The configuration agent selects a length for the server ID encoding that MUST
+have enough entropy to have a different code point for each server. It must be
+no more than seven octets.
+
+The configuration agent also selects an LB Timeout.
 
 ### Load Balancer Actions
 
-On each incoming packet, the load balancer extracts consecutive octets,
-beginning with the second octet. If these octets match an existing server ID,
-the packets are routed to the new server ID.  If the octets are not a match
-and the packet is an Initial packet, then it is assumed to be a new flow and
-load balanced.
+The load balancer maintains a table of all assigned server IDs, which is
+initialized empty. On each incoming packet, the load balancer extracts
+consecutive octets, beginning with the second octet. If these octets match an
+existing server ID, the packets are routed to the new server ID. As there are no
+assigned server IDs at first, the first incoming packet has a non-compliant DCID
+by definition.
+
+The server ID table is distinct for each instance of low-config CID algorithm
+configuration, even if two configurations use the same server ID length. Thus
+a given server ID might correspond to a different server in each configuration.
+
+Note that each server will usually have many server IDs assigned to it.
+
+The load balancer MUST keep track of the most recent observation of each server
+ID in the table and delete the entries when the time since that observation
+exceeds the LB Timeout.
+
+#### Special Rules for Non-compliant DCIDs
+
+In QUIC version 1, incoming Connection IDs generated by clients should be no
+less than 8 octets, so all non-compliant DCIDs should not be due to insufficient
+length. Load balancers that are operating the Low-Config CID algorithm SHOULD
+drop incoming packets with CIDs too short to execute it unless it can be routed using another configured algorithm.
+
+If the DCID is non-compliant solely because the server ID is unassigned, the
+load balancers follows the procedures in {{routing-algorithms}} and adds the
+server ID and assignment the table.
+
+If the DCID references the 4-tuple routing bits or an undefined configuration,
+use the following procedure:
```

Comment:
It might be worth summarizing what the following steps are intended to do.  I think they're intending to use the existing bytes in the DCID as the Server ID with the most recent config?

Reply:
fixed